drawBot/ui/codeEditor.py

```diff
@@ -589,6 +590,32 @@ def keyDown_(self, event):
                 self._insertTextAndRun("True", selectedRange)
                 return
 
+        languageData = self.languagesIDEBehaviorForLanguage_(self.lexer().name)
+        if languageData:
+            autoCloseMap = languageData.get("autoCloseMap", dict())
+            if char in autoCloseMap:
+                selectedText = self.string().substringWithRange_(selectedRange)
+                selectedText = selectedText.strip()
+
+                try:
+                    nextChar = self.string().substringWithRange_((selectedRange.location, 1))
+                    nextChar = nextChar.strip()
+                    if nextChar in autoCloseMap.values():
+                        nextChar = ""
+                except Exception:
```

Comment:
What exactly are you catching here?

Reply:
when the string is smaller then range with the same location and a length of 1

line 602-604 and could be outside the try except

Reply:
So there's two issues here:
- the exception caught is too broad (`Exception` is at the base of the exception hierarchy, so it's pretty much a catch all, potentially masking bugs)
- there are irrelevant lines within the `try`

Reply:
Consider `try: ... except TheExactExceptionToBeCaught: ... else: ...`

Reply:
it seems to be an `IndexError`